helix-term/src/ui/editor.rs

```diff
@@ -808,13 +809,19 @@ impl EditorView {
         //-------------------------------
         // Middle / File path / Title
         //-------------------------------
+        let modified = if doc.is_modified() { "[+]" } else { "" };
+        let readonly = if doc.file_stat == FileStat::ReadOnly {
+            editor.config().read_only_indicator.to_string()
+        } else {
+            String::new()
+        };
         let title = {
             let rel_path = doc.relative_path();
             let path = rel_path
                 .as_ref()
                 .map(|p| p.to_string_lossy())
                 .unwrap_or_else(|| SCRATCH_BUFFER_NAME.into());
-            format!("{}{}", path, if doc.is_modified() { "[+]" } else { "" })
+            format!("{}{}{}", readonly, path, modified)
```

Comment:
```suggestion
            format!("{}{}{}", path, readonly, modified)
```

It looks like it's part of the path if it comes before without any spacing

Reply:
Yeah, that was me playing around with the position of the unicode lock icon. I prefer it before the filename, but it's fine afterwards. I also debated inserting spaces between the path and the indicators, but decided against - it's easy enough to insert a space in the indicator string if you want it.